contracts/randdrop/src/state.rs

```diff
@@ -0,0 +1,50 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Coin, HexBinary, Timestamp, Uint128};
+use cw_storage_plus::{Item, Map};
+
+#[cw_serde]
+pub struct Config {
+    /// manager that can change the manager , register merkle or withdraw funds
+    pub manager: Addr,
+    // The nois-proxy from which to request the randomness
+    pub nois_proxy: NoisProxy,
+    // denom of the token to randdrop
+    pub randdrop_denom: String,
+    /// MerkleRoot is hex-encoded merkle root.
+    pub merkle_root: HexBinary,
+}
+
+#[cw_serde]
+pub struct NoisProxy {
+    // The price to pay the proxy for randomness
+    pub price: Coin,
+    // The address of the nois-proxy contract deployed onthe same chain as this contract
+    pub address: Addr,
+}
+
+#[cw_serde]
+pub struct ParticipantData {
+    // The randomness beacon received from the proxy
+    pub nois_randomness: Option<[u8; 32]>,
```

Comment:
Do we need to store this? We just process it the moment it comes in

Reply:
it's noce for people to be able to reproduce/check by themselves. also the dapp could show the randomness per user and give them a tool to do the offchain simulation

Reply:
okay, sure. But let's use HexBinary as the data type here. `Option<[u8; 32]>` is a list of integers at JSON level, such as `[12,254,1,192,17,...,43,15]` which is longer than the hex.

Reply:
if the whole humanity participates in this randdrop the size of the state reserved for randomness is 260GB. 1person= 1 participant